Adding docs for password reset and account deletion

garethtdavies opened the pull request:
Change-type: patch
Signed-off-by: Gareth Davies gareth@balena.io

alisondavis17 commented on pages/learn/manage/account.md:
```diff
@@ -154,12 +158,21 @@ Now when you log in you will be prompted for the code displayed in your authenti
+If you wish to delete your {{ $names.company.lower }} account, go to your [_Preferences_][prefs] page, and under the _Account Details_ tab, select the _Delete Account_ button. You will need to confirm this action by entering your password. Upon confirmation, the account will be permanently deleted, including all applications and devices.
+
+__Note__ If your account does not have a password set you can request a password using the [password reset instructions][password_reset].
+
```
Add note: If you would also like to request deletion of your data in accordance with GDPR, please refer to the instructions in our [privacy policy](https://www.balena.io/privacy-policy/). 

We may also want to add a note about social logins as needed - cc @shaunmulligan

garethtdavies commented:
@sradevski I think the updating password issue mentioned above was resolved?

I've updated this PR to reflect that. This PR was more for the account deletion stuff - I added the reset password docs due to the earlier issue but I've left the generic instructions in, though it might be overkill and can remove.

sradevski commented on pages/learn/manage/account.md:
```diff
@@ -8,6 +8,10 @@ title: Account management
 
+## Password reset
+
+If you forget your password, you may request to reset it via the [password reset][password-reset] page. Enter the email address associated with your {{ $names.company.lower }} account and if the email address is associated with a {{ $names.company.lower }} account, a password reset link will be sent to that address. Following the link, you will be able to select a new password.
```
NIT: 
```suggestion
If you forget your password, you may request to reset it via the [password reset][password-reset] page. Enter the email address associated with your {{ $names.company.lower }} account and if the email address is associated with a {{ $names.company.lower }} account, a password reset link will be sent to that address. Following the link, you will be able to enter a new password.
```

sradevski commented on pages/learn/manage/account.md:
```diff
@@ -154,12 +158,20 @@ Now when you log in you will be prompted for the code displayed in your authenti
 
+## Delete account
+
+If you wish to delete your {{ $names.company.lower }} account, go to your [_Preferences_][prefs] page, and under the _Account Details_ tab, select the _Delete Account_ button. You will need to confirm this action by entering your password. Upon confirmation, the account will be permanently deleted, including all applications and devices. If you would also like to request deletion of your data in  ...
```
As Alison said, we probably want to specify that if they sign up using social auth, they first need to set a password before being able to delete the account.

sradevski commented:
@garethtdavies Yes, it was resolved, it is good to go.

garethtdavies commented:
@xginn8 @shaunmulligan @chrisys can I get a quick review of this, please 😄

xginn8 commented on pages/learn/manage/account.md:
```diff
@@ -8,6 +8,10 @@ title: Account management
 
+## Password reset
+
+If you forget your password, you may request to reset it via the [password reset][password-reset] page. Enter the email address associated with your {{ $names.company.lower }} account and if the email address is associated with a {{ $names.company.lower }} account, a password reset link will be sent to that address. Following the link, you will be able to enter a new password.
```
```suggestion
If you forget your password, you may request to reset it via the [password reset][password-reset] page. Enter the email address associated with your {{ $names.cloud.lower }} account. If the email address has an associated account, a password reset link will be sent to that address. Following the link, you will be able to enter a new password.
```

xginn8 commented on pages/learn/manage/account.md:
```diff
@@ -154,12 +158,20 @@ Now when you log in you will be prompted for the code displayed in your authenti
 
+## Delete account
+
+If you wish to delete your {{ $names.company.lower }} account, go to your [_Preferences_][prefs] page, and under the _Account Details_ tab, select the _Delete Account_ button. You will need to confirm this action by entering your password. If your account does not have a password, you will be prompted to set one in your account preferences. Upon confirmation, the account will be permanently delet ...
```
```suggestion
If you wish to delete your {{ $names.cloud.lower }} account, go to your [_Preferences_][prefs] page, and under the _Account Details_ tab, select the _Delete Account_ button. You will need to confirm this action by entering your password. If your account does not have a password, you will be prompted to set one in your account preferences. Upon confirmation, the account will be permanently deleted, including all applications and devices. If you would also like to request deletion of your data in accordance with GDPR, please refer to the instructions in our [privacy policy][privacy-policy].
```

garethtdavies commented:
@balena-ci retest
